data-prepper-plugins/geoip-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/GeoIpProviderFactoryTest.java

```diff
@@ -0,0 +1,62 @@
+package com.amazon.dataprepper.plugins.prepper.geoip.provider;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.plugins.prepper.geoip.GeoIPPrepperConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.internal.matchers.apachecommons.ReflectionEquals;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+
+@ExtendWith(MockitoExtension.class)
+public class GeoIpProviderFactoryTest {
+
+    private static final String TEST_DATABASE_PATH = "./src/test/resources/GeoLite2-City-Test.mmdb";
+
+    private PluginSetting pluginSetting;
+
+    private GeoIpProviderFactory factory;
+
+    @BeforeEach
+    void setup() {
+        factory = new GeoIpProviderFactory();
+        pluginSetting = new PluginSetting(
+                "geoip",
+                new HashMap<String, Object>() {{
+                    put(GeoIPPrepperConfig.DATABASE_PATH, TEST_DATABASE_PATH);
+                }}
+        );
```

Comment:
Can also mock `PluginSetting` and force it to return specific values if you find it easier to setup tests with